airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/AutoDisableConnectionActivityImpl.java

```diff
@@ -28,63 +36,131 @@ public class AutoDisableConnectionActivityImpl implements AutoDisableConnectionA
+          disableConnection(input.getConnectionId(), lastJob);
+          return new AutoDisableConnectionOutput(true);
+        }
+
+        // sending warning for disable connection if only failed jobs in the past
+        // maxDaysOfOnlyFailedJobsBeforeWarning days
+        if (numDaysSinceFirstReplicationJob >= maxDaysOfOnlyFailedJobsBeforeWarning &&
+            (successTimestamp.isEmpty()
+                || getDaysSinceTimestamp(currTimestampInSeconds, successTimestamp.get()) >= maxDaysOfOnlyFailedJobsBeforeWarning)) {
+          sendWarningIfNotPreviouslySent(successTimestamp, maxDaysOfOnlyFailedJobsBeforeWarning, firstJob, lastJob, jobs, numFailures);
         }
 
-        final StandardSync standardSync = configRepository.getStandardSync(input.getConnectionId());
-        standardSync.setStatus(Status.INACTIVE);
-        configRepository.writeStandardSync(standardSync);
       } catch (final Exception e) {
         throw new RetryableException(e);
       }
     }
-    return new AutoDisableConnectionOutput(true);
+    return new AutoDisableConnectionOutput(false);
+  }
+
+  private void sendWarningIfNotPreviouslySent(final Optional<Long> successTimestamp,
+                                              final int maxDaysOfOnlyFailedJobsBeforeWarning,
+                                              final Job firstJob,
+                                              final Job lastJob,
+                                              final List<JobWithStatusAndTimestamp> jobs,
+                                              final int numFailures) {
+    if (numFailures > 1 && checkIfWarningPreviouslySent(successTimestamp, maxDaysOfOnlyFailedJobsBeforeWarning, firstJob, jobs)) {
+      return;
+    }
+    jobNotifier.autoDisableConnectionAlertWithoutCustomerioConfig(CONNECTION_DISABLED_WARNING_NOTIFICATION, lastJob);
+  }
+
+  // warning should be previously sent if
+  // 1. no success found in the time span and the previous failure is at least
+  // maxDaysOfOnlyFailedJobsBeforeWarning days after the first job
+  // 2. success found and it is at least maxDaysOfOnlyFailedJobsBeforeWarning days after the previous
+  // failure
+  private boolean checkIfWarningPreviouslySent(final Optional<Long> successTimestamp,
+                                               final int maxDaysOfOnlyFailedJobsBeforeWarning,
+                                               final Job firstJob,
+                                               final List<JobWithStatusAndTimestamp> jobs) {
+    if (jobs.size() <= 1)
+      return false;
+
+    // get previous failed job (skipping first job since that's considered "current" job)
+    JobWithStatusAndTimestamp prevFailedJob = jobs.get(1);
+    for (int i = 2; i < jobs.size(); i++) {
+      if (prevFailedJob.getStatus() == JobStatus.FAILED)
+        break;
+      prevFailedJob = jobs.get(i);
+    }
+
+    return (successTimestamp.isEmpty()
```

Comment:
Similar recommendation here for using local boolean variables to aid comprehension of the nested conditional

Reply:
same as above